### ISSUE TITLE
Constructor property shouldn't be enumerable.

### ISSUE THREAD

**Issue by vivekimsit** [kenwheeler/cash#134]
For example the following snippet lists `constructor` too which is not standard behavior.

```
for (var k in $.prototype) {
 console.log(k);
}
```

Possible fix:

```
Object.defineProperty( $.prototype, "constructor" , {
    enumerable: false,
    writable: true,
    configurable: true,
    value: $    // point `.constructor` at `$`
} );
```

**Comment by shshaw** [comment]
Good catch. Shouldn't `configurable` and `writable` be `false` as well?

**Comment by vivekimsit** [comment]
It depends, but its shouldn't be enumerable for sure.

**Comment by shshaw** [comment]
From [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty): 
> configurable: true if and only if the type of this property descriptor may be changed and if the property may be deleted from the corresponding object.
>
> writable: true if and only if the value associated with the property may be changed with an assignment operator.

Based on that, I'd say they should both be false, which means we're just looking at a simple one-liner since those values are false by default.

```diff
var fn = cash.fn = cash.prototype = Init.prototype = { // jshint ignore:line
-  constructor: cash
  cash: true,
  length: 0,
  push: push,
  splice: ArrayProto.splice,
  map: ArrayProto.map,
  init: Init
};

+ Object.defineProperty(fn,'constructor',{ value: cash });
```

**Comment by vivekimsit** [comment]
LGTM 👍

**Comment by shshaw** [comment]
Fixed in [1.3.5](https://github.com/kenwheeler/cash/releases/tag/1.3.5)
